packages/cli-kit/src/ui.test.ts

```diff
@@ -0,0 +1,53 @@
+import {simpleTask} from './ui.js'
+import {mockAndCaptureOutput} from './testing/output.js'
+import {describe, expect, it, vi} from 'vitest'
+
+describe('simpleTask()', () => {
+  it('outputs correctly using success property when successful', async () => {
+    // Given
+    const title = 'Starting the task'
+    const success = 'Finished the task'
+    const task = vi.fn(() => Promise.resolve())
+    const outputMock = mockAndCaptureOutput()
+    outputMock.clear()
+
+    // When
+    await simpleTask({title, success, task})
+
+    // Then
+    expect(outputMock.started()).toMatch('Starting the task')
+    expect(outputMock.completed()).toMatch('Finished the task')
+    expect(outputMock.failed()).not.toBeDefined()
+  })
+  it('outputs correctly using result of task when successful', async () => {
+    // Given
+    const title = 'Starting the task'
+    const task = vi.fn(() => Promise.resolve('Finished the task (created X and Y)'))
+    const outputMock = mockAndCaptureOutput()
+    outputMock.clear()
+
+    // When
+    await simpleTask({title, task})
+
+    // Then
+    expect(outputMock.started()).toMatch('Starting the task')
```

Comment:
I'd recommend using `toMatchInlineSnapshot` to make it easier to re-record the value we check against in the future if the content/format changes.

Reply:
This comment applies to all the string `toMatch` in this test module.